core/src/main/java/be/isach/ultracosmetics/cosmetics/particleeffects/ParticleEffectRainbowWings.java

```diff
@@ -0,0 +1,142 @@
+                        );
+                    }
+                    location.subtract(v2);
+                    location.subtract(v);
+                }
+                x += space;
+            }
+            y -= space;
+            x = defX;
+        }
+    }
+
+    public static Vector rotateAroundAxisY(Vector v, double angle) {
+        double x, z, cos, sin;
+        cos = Math.cos(angle);
+        sin = Math.sin(angle);
+        x = v.getX() * cos + v.getZ() * sin;
+        z = v.getX() * -sin + v.getZ() * cos;
+        return v.setX(x).setZ(z);
+    }
+
+    public static Vector getBackVector(Location loc) {
+        final float newZ = (float) (loc.getZ() + (1 * Math.sin(Math.toRadians(loc.getYaw() + 90))));
+        final float newX = (float) (loc.getX() + (1 * Math.cos(Math.toRadians(loc.getYaw() + 90))));
+        return new Vector(newX - loc.getX(), 0, newZ - loc.getZ());
+    }
+
+    /**
+     * Generates a color in the rainbow sequence based on the current time.
+     *
+     * @return a Particles.OrdinaryColor representing the color for the current time.
+     */
+    public static Particles.OrdinaryColor getColor() {
+        long currentEpochMilliseconds = System.currentTimeMillis();
+        long currentEpochSeconds = currentEpochMilliseconds / 1000L;
+
+        double secondsPerPhase = 2;
+        double phase = currentEpochSeconds % (6 * secondsPerPhase);
+        phase = phase / secondsPerPhase;
+
+        double millisecondsToAdjustBy = currentEpochMilliseconds % (secondsPerPhase * 1000);
+
+        // There's a change of a specific number of rgb values in a single tick. This calculation is basically the
+        // current tick (or milliseconds) multiplied by 255 to get the rgb value.
+        int currentRotatingRgbValue = (int) (millisecondsToAdjustBy * 0.255 / secondsPerPhase);
+
+        int red = 0;
+        int green = 0;
+        int blue = 0;
+
+        // There are six phases in a rainbow when adjusting rgb values. This if-else encapsulates those six phases.
+        if (phase >= 0 && phase < 1) { // Red is at 255, Green goes to 255
+            red = 255;
+            green = currentRotatingRgbValue;
+        } else if (phase >= 1 && phase < 2) { // Red goes to 0
+            red = 255 - currentRotatingRgbValue;
+            green = 255;
+        } else if (phase >= 2 && phase < 3) { // Blue goes to 255
+            green = 255;
+            blue = currentRotatingRgbValue;
+        } else if (phase >= 3 && phase < 4) { // Green goes to 0
+            green = 255 - currentRotatingRgbValue;
+            blue = 255;
+        } else if (phase >= 4 && phase < 5) { // Red goes to 255
+            red = currentRotatingRgbValue;
+            blue = 255;
+        } else if (phase >= 5 && phase < 6) { // Blue goes to 0
+            red = 255;
+            blue = 255 - currentRotatingRgbValue;
+        }
+
+        Bukkit.getServer().getLogger().warning("currentSeconds: " + currentEpochSeconds);
```

Comment:
```suggestion
```
Forgotten debug message? This actually prevents it from compiling for me since Bukkit isn't imported

Reply:
Yeah, sorry! Just removed it!